src/ast/ast.c

```diff
@@ -559,10 +560,15 @@ void AST_Free(AST *ast) {
 
 cypher_parse_result_t *parse_query(const char *query) {
 	FILE *f = fmemopen((char *)query, strlen(query), "r");
-	cypher_parse_result_t *result = cypher_fparse(f, NULL, NULL, CYPHER_PARSE_ONLY_STATEMENTS);
+	cypher_parse_result_t *result = cypher_fparse(f, NULL, NULL, CYPHER_PARSE_SINGLE);
 	fclose(f);
 
 	if(!result) return NULL;
+	
+	if(!cypher_parse_result_eof(result)) {
+		ErrorCtx_SetError("Error: query is incomplete.");
```

Comment:
this is a valid cypher query `MATCH (a) RETURN a; CREATE();`
which we reply with: `Error: query is incomplete.` please consult with @jeffreylovitz to find a more suitable error message.